packages/react/src/collapsible/panel/useCollapsiblePanel.ts

```diff
@@ -77,10 +77,14 @@ export function useCollapsiblePanel(
 
   useEnhancedEffect(() => {
     setPanelId(id);
+
+    if (!keepMounted && !open) {
+      setPanelId(undefined);
+    }
```

Comment:
No need to set the state twice:

```suggestion
    if (!keepMounted && !open) {
      setPanelId(undefined);
    } else {
      setPanelId(id);
    }
```

Reply:
@michaldudak Fixed ~